### ISSUE TITLE
Problems using `agent-services` on server

### ISSUE THREAD

**Issue by mmiller-max** [allegroai/clearml-server#113]
Hi, I've had a three small issues running `agent-services` via `docker-compose up agent-services` on the server, so thought I'd put them in one issue.

###  `ERROR: Invalid requirement: 'clearml-agent">=0.17.0"'`

I think this comes from `docker-compose.yml`

https://github.com/allegroai/clearml-server/blob/04ea9018a3c0cbb8a9e0044e5a24110883a0abfa/docker/docker-compose.yml#L139

should be

```
CLEARML_AGENT_UPDATE_VERSION: ${CLEARML_AGENT_UPDATE_VERSION:->=0.17.0}
```

If you're happy with this change I can make a quick PR.

### When starting `agent-services` at the same time as `apiserver`, `apiserver` is not ready quickly enough

When running `docker-compose up` for all services at once, I see this error with `agent-services`.

```
clearml-agent-services | Using environment access key CLEARML_API_ACCESS_KEY=****
clearml-agent-services | Using environment secret key CLEARML_API_SECRET_KEY=********
clearml-agent-services | 
clearml-agent-services | clearml_agent: ERROR: Connection Error: it seems *api_server* is misconfigured. Is this the ClearML API server http://apiserver:8008 ?
clearml-agent-services | 
clearml-agent-services exited with code 1
```

This error doesn't happen if I manually wait for the server to be ready.

Is it possible to add some sort of retrying to get around this? For example adding this to the docker-compose helps (although I realise in other situations it may not be desirable to repeat indefinitely, e.g. if `CLEARML_API_ACCESS_KEY` not supplied)

```
restart: on-failure
```

### Default SlackAlert fails

The requirement for the slack client that is in the `Slack Alert` task is `slackclient==2.8.0`. However this leads to an error when running `from slack_sdk import WebClient` as `slack_sdk` doesn't exist (it was previously just `from slack import WebClient`). I manually got around it by changing the requirements so that it installed `slack_sdk==3.15.1` which solves the issue.

**Comment by mmiller-max** [comment]
>Thanks for taking the time to help us improve ClearML 

All good!

> That would be great! 

Here you go: #114 

> I think perhaps a better solution is adding some kind of simple bash script probe (waiting until the server responds to `debug.ping` or something similar) to the `agent-services` entrypoint (we can even override the entrypoint in the docker-compose file and do it there since doing it in the entrypoint itself will be too far removed from the docker-compose itself) - WDYT?

Yep I think something like this sounds good. Where is the current entrypoint/`Dockerfile` for agent-services? I can't see where to edit it!

> A PR fixing this required version will be appreciated as well 

Likewise I'm not sure where I can edit this - it's the task that comes packaged with the server (in my case the GCP disk image) that needs updating. The script itself is in the normal `clearml` repository but I don't think that includes requirements. Please can you point me to the right place?

**Comment by jkhenning** [comment]
> Here you go: https://github.com/allegroai/clearml-server/pull/114

Thanks 🙂 

> Yep I think something like this sounds good. Where is the current entrypoint/Dockerfile for agent-services? I can't see where to edit it!

The current entrypoint is in https://github.com/allegroai/clearml-agent/blob/master/docker/services/entrypoint.sh
However, I really don't think you should edit the entrypoint, but change the command in the docker compose file and launch the entrypoint.sh file once the server pings back

> Likewise I'm not sure where I can edit this

As far as I know this script is based on the example code in https://github.com/allegroai/clearml/tree/master/examples/services/monitoring - am I wrong?

**Comment by mmiller-max** [comment]
> However, I really don't think you should edit the entrypoint, but change the command in the docker compose file and launch the entrypoint.sh file once the server pings back

Yep I agree! Just wanted to check what was being used as the entrypoint so I can launch it after the ping

> As far as I know this script is based on the example code in https://github.com/allegroai/clearml/tree/master/examples/services/monitoring - am I wrong?

I didn't see the requirements file next to the script! 😁

**Comment by mmiller-max** [comment]
>  think perhaps a better solution is adding some kind of simple bash script probe (waiting until the server responds to `debug.ping` or something similar)

What do you think is the best way to check that the server is ready? I tried curling `http://apiserver:8008` but wasn't sure if a response from this meant it was actually ready or just that the http routing had started.

**Comment by jkhenning** [comment]
`curl http://apiserver:8008/debug.ping` should do it - if you're getting `200` it means the server is up and running

**Comment by mmiller-max** [comment]
Closed by #113

**Comment by mmiller-max** [comment]
Sorry forgot there was more than one thing in this issue, also needs #114

**Comment by mmiller-max** [comment]
Closed by #114

**Comment by mmiller-max** [comment]
Closed by #114
